utils/rule_dir_json.py

```diff
@@ -46,13 +53,31 @@ def walk_products(root, all_products):
         guide_dir = os.path.join(product_dir, product_yaml['benchmark_root'])
         guide_dir = os.path.abspath(guide_dir)
 
+        product_to_guide[product] = guide_dir
+        if guide_dir not in guide_to_products:
+            guide_to_products[guide_dir] = set()
+        guide_to_products[guide_dir].add(product)
```

Comment:
What about using collections.defaultdict for guide_to_products?

Reply:
I keep forgetting it is allowed in Python 2... Even though it was already imported. :D